Makefile

```diff
@@ -5,7 +5,7 @@ format-check:
 format:
 	black .
 typecheck:
-	mypy -p rich --strict --no-incremental
+	pre-commit run --all-files mypy
 typecheck-report:
```

Comment:
This `typecheck-report`) is not used anywhere. Can it be removed? Then `mypy` could be dropped for the dev deps, which would simplify the install/update process in poetry.

Reply:
What would be simplified exactly?

I use it to generate typecheck reports.

Reply:
My comment is not very valid anymore anyway - I usually recommend running mypy via pre-commit only; mypy is very sensitive to its environment, and it is very sensitive to the target Python version (sometimes you can't make it pass on multiple versions due to type ignore's changing validity based on Python version - we gave up in `pypa/build`, IIRC). But this running a different way, and passing, so not going to suggest anything different for now.